Client/src/app/_services/account.service.ts

```diff
@@ -26,12 +49,48 @@ export class AccountService {
       }));
   }
 
-  setCurrentUser(user: User) {
+  public logout() {
+    localStorage.removeItem('user');
+    this.currentUser.next({});
+  }
+
+  public isAuthorised(user?: User): boolean {
+    if (!user) {
+      user = this.currentUser.getValue();
+    }
+    if (user && user.token) {
+      return this.checkTokenTime(user.token);
+    }
+    return false;
+  }
+
+  public getAccessToken(): string | undefined {
+    return this.currentUser.getValue()?.token;
+  }
+
+  private checkTokenTime(token: string | undefined): boolean {
+    if (token) {
+      const mainPart = token.split('.')[1];
+      const decodedString = atob(mainPart);
+      const expirationTime = JSON.parse(decodedString)['exp'] * 1000;
+      const now = Date.now();
+      return now < expirationTime;
+    }
+    return false;
+  }
+
+  private setCurrentUser(user: User) {
     this.currentUser.next(user);
   }
 
-  logout() {
-    localStorage.removeItem('user');
-    this.currentUser.next(undefined);
+  private handleError(error: HttpErrorResponse, showError: (message: string) => void) {
```

Comment:
We should also make use of interceptors, https://angular.io/api/common/http/HttpInterceptor that way we can centralize our exception logic and if needed navigate the user to a "error page". This might be a good way to introduce a package such as toastr for some smooth notifications

Reply:
@chrisK00 I like the idea. Done.